src/IO/ADIOS/ADIOS2IOHandler.cpp

```diff
@@ -2032,7 +2035,18 @@ namespace detail
         }
         else
         {
-            var.SetShape(shape);
```

Comment:
I don't think that the previous `shape` should be considered here at all.
This code is executed when the Variable was already part of the previous step. The Variable shape can be something completely new in the new step. If the user means to use a joined dimension again, `Dataset::JOINED_DIMENSION` should be specified again; we should not try to outsmart the user by trying to keep the joined dimension from the previous step.